app/javascript/app/providers/ghg-target-emissions-provider/ghg-target-emissions-provider-actions.js

```diff
@@ -0,0 +1,31 @@
+import { createAction, createThunkAction } from 'redux-tools';
+import { INDOAPI } from 'services/api';
+
+export const fetchGHGTargetEmissionsInit = createAction(
+  'fetchGHGTargetEmissionsInit'
+);
+export const fetchGHGTargetEmissionsReady = createAction(
+  'fetchGHGTargetEmissionsReady'
+);
+export const fetchGHGTargetEmissionsFail = createAction(
+  'fetchGHGTargetEmissionsFail'
+);
+
+export const fetchGHGTargetEmissions = createThunkAction(
+  'fetchGHGTargetEmissions',
+  params => (dispatch, state) => {
+    const { GHGTargetEmissions } = state();
+    if (!GHGTargetEmissions.loading) {
```

Comment:
I think it's also worth checking if the data is already fetched so prevent re-fetching

Reply:
I'm checking in the action if the params are the same. The params change the data we store in this case so we can't prevent the fetching